napari/_qt/qt_viewer.py

```diff
@@ -655,78 +655,71 @@ def on_resize(self, event):
         """
         self.viewer._canvas_size = tuple(self.canvas.size[::-1])
 
-    def on_mouse_wheel(self, event):
-        """Called whenever mouse wheel activated in canvas.
-
+    def _process_mouse_event(self, mouse_callbacks, event):
+        """Called whenever mouse pressed in canvas.
         Parameters
         ----------
-        event : qtpy.QtCore.QEvent
+        mouse_callbacks : function
+            Mouse callbacks function.
+        event : vispy.event.Event
+            The vispy event that triggered this method.
         """
         if event.pos is None:
             return
 
-        event = ReadOnlyWrapper(event)
+        # Update the cursor position
         self.viewer.cursor.position = self._map_canvas2world(list(event.pos))
-        mouse_wheel_callbacks(self.viewer, event)
+
+        # Add the cursor position to the event
+        event.position = self.viewer.cursor.position
```

Comment:
mentioned in summary, can this be `event.pos_world`?

Reply:
I see how this is confusing having both `event.pos` given by vispy which is position in canvas coordinates and `event.position` which is now our position in world coordinates, but I like the similarity between `viewer.cursor.position` and `event.position` which now are the same. I'd much rather rename `event.pos` to `event.canvas_position` or `event.position_canvas` at which point having `event.position` I think is ok. I also didn't and don't plan to put the `data_position` on the event (I had explored this in other PRs, but don't think it is needed).

My preference would be to stick with `event.position` for this property, which I think will be the most commonly used one, and then gradually rename some of the vispy ones as we make our own event here in followup PRs.

Reply:
I'd probably vote for event.pos_canvas, and event.pos_world and _neither_ event.pos or event.position.   But I'm definitely ok with whatever you and @jni ultimately choose here.

Reply:
Oof. Tough one. I love brevity but I do think it's good to be clear here on which position is which. I *still I do like @tlambert03's suggestion of specifying *both*, `event.pos_canvas` and `event.pos_world` or `event.position_canvas` and `event.position_world`. What do you think of that suggestion @sofroniewn?

fwiw I'm not sure I agree that world coordinates will be the most common use — wouldn't you think that many people will want to access the data based on the cursor?

Reply:
oops, I just realised canvas != data. Gonna be a long week LOL

Reply:
ok, given the above, I'm happy with just `event.position` for now, with the plan to rename `.pos` to `.position_canvas`.

Reply:
Thanks for all the input here, I will leave as is for now, but will flag in issue as something to do